server/src/controllers/apps.controller.ts

```diff
@@ -197,8 +199,8 @@ export class AppsController {
     } else {
       apps = await this.appsService.all(user, page, searchKey);
     }
-    //remove password from user info
-    apps.forEach((app) => (app.user.password = undefined));
+    //remove user sensitive data
+    apps.forEach((app) => this.userService.removeSensitiveData(app.user));
```

Comment:
Shall we try changing the query?
```
const viewableAppsQb = createQueryBuilder(App, 'apps')
      .select(['app_group_permissions', 'user.first_name', 'user.last_name'])
      .innerJoin('apps.groupPermissions', 'group_permissions')
      .innerJoin('apps.appGroupPermissions', 'app_group_permissions')
      .innerJoin('apps.user', 'user')
      .innerJoin(
        UserGroupPermission,
        'user_group_permissions',
        'app_group_permissions.group_permission_id = user_group_permissions.group_permission_id'
      )
      .where('apps.organization_id = :organizationId', { organizationId: user.organizationId })
      .andWhere(
        new Brackets((qb) => {
          qb.where('user_group_permissions.user_id = :userId', {
            userId: user.id,
          })
            .andWhere('app_group_permissions.read = :value', { value: true })
            .orWhere('apps.is_public = :value OR apps.user_id = :userId', {
              value: true,
              userId: user.id,
            });
        })
      );
    if (searchKey) {
      viewableAppsQb.andWhere('LOWER(apps.name) like :searchKey', {
        searchKey: `%${searchKey && searchKey.toLowerCase()}%`,
      });
    }
```
[Reference](https://github.com/typeorm/typeorm/issues/876)